### PR TITLE
Add profile page with achievements and statistics

### PR DESCRIPTION
**Pull request by carldouch**
### Add ProfilePage, Achievements, and Statistics Components

- Added a button in the header and mobile menu for navigation.
- Implemented Achievements section to display user accomplishments.
- Added Statistics section with filter options: Today, This Month, This Year, and All-time.
  - **Overview**: Summary of statistics.
  - **Charts**: Visual representation of data.
  - **Details**: In-depth statistics.
- Introduced `started_time` in `useGame` for tracking game start times.

## Summary by CodeRabbit

- **New Features**
	- Introduced a `ProfilePage` component for user profile management within the Mobile Menu.
	- Added a new `Achievements` component to display achievement alerts based on game performance.
	- Implemented a `Statistics` component for visualizing and analyzing game statistics.
	- Integrated a charting component using the Recharts library for enhanced data visualization.
	- Added alert components for contextual notifications in the UI.

- **Enhancements**
	- Improved the `Game` class to track the start time of games.
	- Enhanced the user interface with new CSS animations and scrollbar styles.

- **Bug Fixes**
	- Resolved minor formatting issues in the `package.json` for better readability.

### CODE REVIEW

File: client/src/ui/modules/ProfilePage.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+import { Statistics } from "./Statistics";
+import Connect from "../components/Connect";
+import { useGames } from "@/hooks/useGames";
+import { usePlayer } from "@/hooks/usePlayer";
+import useAccountCustom, { ACCOUNT_CONNECTOR } from "@/hooks/useAccountCustom";
+import { Tabs, TabsContent, TabsList, TabsTrigger } from "@/ui/elements/tabs";
+import { motion } from "framer-motion";
+import { Game } from "@/dojo/game/models/game";
+
+// Constantes pour les seuils de score et de combo
```
Comment (Cheelax) [inline client/src/ui/modules/ProfilePage.tsx:21]: Traduis les coms en anglais stp

File: client/src/ui/modules/Achievements.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import { Game } from "@/dojo/game/models/game";
+import { useTheme } from "@/ui/elements/theme-provider";
+import getElementStyle from "../theme/GetElementStyle";
+
+interface AchievementsProps {
+  gamesByThresholds: { threshold: number; games: Game[] }[];
+  combosByThresholds: { threshold: number; games: Game[] }[];
+}
+
+const AchievementAlert: React.FC<{
```
Comment (Cheelax) [inline client/src/ui/modules/Achievements.tsx:13]: Can be worth to put AchevementAlert elsewhere

File: client/src/ui/modules/ProfilePage.tsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+const SCORE_THRESHOLDS = [50, 100, 150, 200, 250, 300, 350, 400, 450, 500];
+const COMBO_THRESHOLDS = [5, 10, 20, 30, 40, 50, 60, 70, 80, 90, 100];
```
Comment (Cheelax) [inline client/src/ui/modules/ProfilePage.tsx:23]: Those variables are already implemented here: client/src/dojo/game/types/level.ts
use them instead if you can

File: client/src/ui/modules/Statistics.tsx [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+  faFire,
+  faWebAwesome,
+  faStar,
+  faCalendar,
+  faArrowsAltH,
+} from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { Game } from "@/dojo/game/models/game";
+
+// Fonction utilitaire pour formater les dates
```
Comment (Cheelax) [inline client/src/ui/modules/Statistics.tsx:51]: english comments

File: client/src/ui/modules/ProfilePage.tsx (hunk 1 above)
Comment (carldouch) [reply client/src/ui/modules/ProfilePage.tsx:21]: Yep

File: client/src/ui/modules/Achievements.tsx (hunk 2 above)
Comment (carldouch) [reply client/src/ui/modules/Achievements.tsx:13]: Yes, created new compo for it

File: client/src/ui/modules/ProfilePage.tsx (hunk 3 above)
Comment (carldouch) [reply client/src/ui/modules/ProfilePage.tsx:23]: Alright.
For the combo, I'll set a step of 5.

File: client/src/ui/modules/Statistics.tsx (hunk 4 above)
Comment (carldouch) [reply client/src/ui/modules/Statistics.tsx:51]: yep

File: client/src/ui/modules/Statistics.tsx [hunk 5]
```diff
@@ -0,0 +1,342 @@
...
+      calculateBonusUsage(filteredGames);
+    return [
+      { bonusType: "Hammer", count: hammerTotal, use: hammerUsed },
+      { bonusType: "Wave", count: waveTotal, use: waveUsed },
+      { bonusType: "Totem", count: totemTotal, use: totemUsed },
+    ];
+  }, [games, selectedPeriod]);
+
+  const chartData = [
+    { month: "January", desktop: 186 },
```
[the remark this answers is not in the document]
Comment (carldouch) [reply client/src/ui/modules/Statistics.tsx:160]: copy paste glitch

**Pull request merged**